_pytest/terminal.py

```diff
@@ -134,16 +134,21 @@ def __init__(self, config, file=None):
         self.showfspath = self.verbosity >= 0
         self.showlongtestinfo = self.verbosity > 0
         self._numcollected = 0
+        self._session = None
 
         self.stats = {}
         self.startdir = py.path.local()
         if file is None:
             file = sys.stdout
         self._writer = _pytest.config.create_terminal_writer(config, file)
+        self._screen_width = self.writer.fullwidth
```

Comment:
I had to cache this otherwise I noticed a good 20% performance penalty. We probably need to fix this in `py` somehow, which we have tried to do in the past.